---
title: "CODE"
author: "Team 04 Group 04"
date: '2022-07-12'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


Loading libraries
```{r}
library(affy)
library(vsn)
library(AnnotationDbi)
library(moe430ammenstcdf)
library(moe430ammenstprobe)
library(cluster)
library(dplyr)
library(hexbin)
library(pheatmap)
library(rstudioapi)
library(tidyverse)
library(readr)
library(ggplot2)
library(knitr)
library(limma)
library(Rcpp)
library(icd.data)
library(tibble)
```

Setting project path
```{r}
projectPath <- dirname(rstudioapi::getSourceEditorContext()$path)
```

Setting working directory and loading mouse data
```{r}
setwd(paste(projectPath,"rawdata","mouse_raw_18",sep = "/"))
data.mouse=ReadAffy()
data.mouse@cdfName <- "MOE430A_Mm_ENST"
```

1. Quality Control
1.1 Individual chips
```{r}
for (i in 1:18){
   image(data.mouse[,i], col = rainbow (100, start = 0, end = 0.75)[100:1])}
# chips 6,1,4,15 have a low quality
```

1.2 RNA degradation
```{r}
mouse_RNAdeg <- AffyRNAdeg(data.mouse)
plotAffyRNAdeg(mouse_RNAdeg, col = rainbow(15))
```

1.3 Normalization and mean - Sd plot
```{r}
set.seed(234)
vsnmouse <- vsnrma(data.mouse)
meanSdPlot(vsnmouse, plot = FALSE)$gg + theme(aspect.ratio = 1)
vsn.mouse = exprs(vsnmouse)
```

1.4 Boxplots
```{r}
## Before normalization
par ( las =2)
boxplot (data.mouse ,col = rainbow (150),cex.axis = 0.4,main ="Gene expression mouse embryonic development before normalization")

## After normalization
par ( las =2)
boxplot (vsn.mouse,col = rainbow (150),cex.axis =0.5,main ="Gene expression mouse embryonic development after normalization")
```

1.5 Histograms
```{r}
## Before normalization
par(mai = c(0.9,0.9,0.9,0.5))
hist(data.mouse, col = rainbow(10), main = "Density function of log intensity mouse ED before normalization")

## After normalization
par(mai = c(0.9,0.9,0.9,0.5))
plot(density(vsn.mouse), type = "n", xlab = "log Intensity", ylim = c(0, 0.6),main = "Density function of log intensity mouse ED after normalization")
for (i in 1:ncol(vsn.mouse)){lines(density(vsn.mouse[,i]), col = rainbow(10)[i])}
```
2. Cleaning the raw mouse data
2.1 Loading and normalizing the 15 good quality chips
```{r}
#setwd(paste(projectPath,"rawdata","qualityzipped",sep = "/"))
#qdata.mouse = ReadAffy()
#qdata.mouse@cdfName <- "MOE430A_Mm_ENST"
#vsn.qmouse = exprs(vsnrma(qdata.mouse))
#data_ohneaffx= vsn.qmouse[65:39345,]
#namen=(substr(rownames(data_ohneaffx),0,18)[1:39281])
#rownames(data_ohneaffx) <- c(namen)
load("normalized_data.rda")
```

2.2 PCA
```{r}
pca = prcomp(t(data_ohneaffx), scale = TRUE)
pca_var = pca$sdev^2    #calculating variation
pca_var_per = round(pca_var/sum(pca_var)*100,1)   # variation into percentage
barplot(pca_var_per, main = "Scree plot", xlab ="Principal Component", ylab ="Percent Variation")
pca_data = data.frame(Sample = rownames(pca$x), X= pca$x[,1], Y = pca$x[,2])
ggplot(data=pca_data, aes(x=X, y=Y, label=Sample)) +
  xlab(paste("PC1 - ", pca_var_per[1], "%", sep="")) +
  ylab(paste("PC2 - ", pca_var_per[2], "%", sep="")) +
  theme_bw() +
  ggtitle("PCA mouse data") +
  geom_text(col= c("red","red","red","blue","blue","black","black","black","gray","gray","gray","green","purple","purple","purple"))
```
3.TRA matching and differential gene expression (DGE) of mouse data
3.1 Reading in both TRA datasets and joining them together
```{r}
#setwd(paste(projectPath, "rawdata", "TRA_data", sep = "/"))
#TRAs1 <-read_tsv("tra.2014.mouse.5x.table.tsv")
#TRAs2 <-read_tsv("tra.2014.mouse.4301.5x.table.tsv")
#TRA_total = rbind(TRAs1,TRAs2)
#unique_TRAs = unique(TRAs1$ensembl.transcript, TRAs2$ensembl.transcript)
#TRAs_mousedataset = data_ohneaffx[which(rownames(data_ohneaffx) %in% unique_TRAs),] #removed duplicates
#save.image(file="TRAs_mousedata.rda")
load("TRAs_mousedata.rda")
```

3.2 Differential gene expression (DGE)
```{r}
design.m= model.matrix(~0+factor(c(1,1,1,2,2,3,3,3,4,4,4,5,6,6,6)))
colnames(design.m)= c("one", "two","four", "eight", "morula", "blastocyst")
fit.m= lmFit(TRAs_mousedataset, design.m)
contrast.matrix.m= makeContrasts(two-one,four-two, eight-four, morula-eight, blastocyst-morula,levels = design.m)
fit2.m= contrasts.fit(fit.m, contrast.matrix.m)
fit2.m= eBayes(fit2.m)
#save(fit2.m,file = "~/GitHub/project/fit_matched2.RData" )     what is this??
topTable.m = topTable(fit2.m, coef=1)
#linear model is done
#write.fit(fit2.m, file="fit_mouse.csv")
dge_df.m =topTable(fit2.m, coef=1, number=nrow(fit2.m))
results.m= decideTests(fit2.m)
#should i put summary and view here?
```

3.3 Visualizing results of DGE
3.3.1 Barplots of (de)regulated genes vs cell stadium
```{r}
barplot(summary(results.m)[1,],las =2, ylab="sum of downregulated genes")
barplot(summary(results.m)[2,],las =2, ylab="sum of  genes with no significant change")
barplot(summary(results.m)[3,],las =2, ylab="sum of upregulated genes")
```
VOLCANO PLOT??????
3.3.2 Barplots of TRA frequency in different tissues
BRAUCHEN WIR DAS?
```{r}
TRA_tissuenames = sort(table(c(as.data.frame(TRA_total[,11]))))
barplot(TRA_tissuenames,las =2, xlab = "tissue" , ylab="Frequency", main ="Frequency of TRAs in all tissues")
barplot(tail(TRA_tissuenames),las =2, xlab = "tissue" , ylab="Frequency",main ="Frequency of TRAs in top tissues")
```

3.3.3 Barplots of TRA frequency in different tissues (only TRAs found in our mouse data)
```{r}
tissue = TRA_total[is.element(TRA_total$ensembl.transcript,rownames(TRAs_mousedataset)),]
matched_tissue = tissue[!duplicated(tissue[,1]),]
matched_tissuenames = sort(table(c(as.data.frame(matched_tissue[,11]))))
barplot(matched_tissuenames, las =2, xlab = "tissue" , ylab="Frequency",main ="Frequency of matched TRAs in all tissues")
barplot(tail(matched_tissuenames), las =2, xlab = "tissue" , ylab="Frequency", main = "Frequency of matched TRAs in top tissues")
```

3.4 
```{r}
#DGE_tissues = cbind(results.m, matched_tissue[,11])
#DGE_foureight = cbind(DGE_tissues[,3],DGE_tissues[,6])
#rownames(DGE_foureight) = rownames(DGE_tissues)
#colnames(DGE_foureight) = c("four to eight","max.tissue")
#DGE_foureight_oe = DGE_foureight[DGE_foureight[,1] == 1,]
#DGE_foureight_oe_count = sort(table(c(as.data.frame(DGE_foureight_oe[,2]))))
setwd(paste(projectPath,"RDS_Files",sep = "/"))
DGE_foureight_oe_count = readRDS(file="DGE_foureight_oe_count.RDS")
####should i name it sth else???
barplot(DGE_foureight_oe_count,las =2, xlab = "tissue" , ylab="Frequency",main ="Frequency of matched TRAs that are overexpressed")
####xlab looks weird
barplot(tail(DGE_foureight_oe_count),las =2, xlab = "tissue" , ylab="Frequency",main="Frequency of matched TRAs that are overexpressed in the top 6 tissues")
```

DGE_foureight_oe_top = tail(DGE_foureight_oe_count)
barplot(DGE_foureight_oe_count,las =2, xlab = "tissue" , ylab="Frequency",main ="Frequency of matched TRAs that are overexpressed")
view(DGE_foureight_oe_count)
barplot(DGE_foureight_oe_top,las =2, xlab = "tissue" , ylab="Frequency",main="Frequency of matched TRAs that are overexpressed in the top 6 tissues")
# We conclude that the blastocyst has the highest overexpression
```